### PR TITLE
Add support for setting specific shm for a resource

### PR DESCRIPTION
**Pull request by modfinmarwi**

### CODE REVIEW

**Comment by J-Zeitler** [comment]
awesome! hopefully

**Comment by modfinmarwi** [comment]
@larsve FYI

File: internal/cloning/cloning.go [hunk 1]
```diff
@@ -154,6 +158,7 @@ func createClone(dss *zfs.Dataset, owner string, snap string, r *internal.Resour
 			Name:              "unless-stopped",
 			MaximumRetryCount: 0,
 		},
+		ShmSize: shmSize,
```
Comment (larsve) [inline internal/cloning/cloning.go:161]: Why not just use this?
```suggestion
		ShmSize: r.Docker.Shm,
```

Comment (modfinmarwi) [reply internal/cloning/cloning.go:161]: It won't be set for most resources - but I guess the default can be moved to when we parse the config?

Comment (larsve) [reply internal/cloning/cloning.go:161]: Before this PR, `ShmSize` in the `container.HostConfig` were zero, not sure what a sane default should be, but I think that even if we set it to zero now, that would be that same as it were before.

Comment (modfinmarwi) [reply internal/cloning/cloning.go:161]: Yes good point - have fixed this in a separate PR

**Pull request merged**